modules/nomad-cluster/main.tf

```diff
@@ -1,44 +1,46 @@
-# THESE TEMPLATES REQUIRE TERRAFORM VERSION 0.8 AND ABOVE
-# ---------------------------------------------------------------------------------------------------------------------
-
+# ----------------------------------------------------------------------------------------------------------------------
+# REQUIRE A SPECIFIC TERRAFORM VERSION OR HIGHER
+# This module has been updated with 0.12 syntax, which means it is no longer compatible with any versions below 0.12.
+# ----------------------------------------------------------------------------------------------------------------------
 terraform {
-  required_version = ">= 0.9.3"
+  required_version = ">= 0.12"
 }
 
 # ---------------------------------------------------------------------------------------------------------------------
 # CREATE AN AUTO SCALING GROUP (ASG) TO RUN NOMAD
 # ---------------------------------------------------------------------------------------------------------------------
 
 resource "aws_autoscaling_group" "autoscaling_group" {
-  launch_configuration = "${aws_launch_configuration.launch_configuration.name}"
-
-  name                = "${var.asg_name}"
-  availability_zones  = ["${var.availability_zones}"]
-  vpc_zone_identifier = ["${var.subnet_ids}"]
-
-  min_size             = "${var.min_size}"
-  max_size             = "${var.max_size}"
-  desired_capacity     = "${var.desired_capacity}"
-  termination_policies = ["${var.termination_policies}"]
-
-  health_check_type         = "${var.health_check_type}"
-  health_check_grace_period = "${var.health_check_grace_period}"
-  wait_for_capacity_timeout = "${var.wait_for_capacity_timeout}"
-
-  tags = [
-    {
-      key                 = "Name"
-      value               = "${var.cluster_name}"
-      propagate_at_launch = true
-    },
-    {
-      key                 = "${var.cluster_tag_key}"
-      value               = "${var.cluster_tag_value}"
-      propagate_at_launch = true
-    },
-    "${var.tags}",
-  ]
+  launch_configuration = aws_launch_configuration.launch_configuration.name
+
+  name                = var.asg_name
+  availability_zones  = var.availability_zones
+  vpc_zone_identifier = var.subnet_ids
+
+  min_size             = var.min_size
+  max_size             = var.max_size
+  desired_capacity     = var.desired_capacity
+  termination_policies = [var.termination_policies]
+
+  health_check_type         = var.health_check_type
+  health_check_grace_period = var.health_check_grace_period
+  wait_for_capacity_timeout = var.wait_for_capacity_timeout
+
+  tags = flatten(
+    [
+      {
+        key                 = "Name"
+        value               = var.cluster_name
+        propagate_at_launch = true
+      },
+      {
+        key                 = var.cluster_tag_key
+        value               = var.cluster_tag_value
+        propagate_at_launch = true
+      },
+      var.tags,
```

Comment:
This is not right. I think you'll end up with a list that contains two maps and then a nested list. See the `for_each` logic in the Vault, Consul, Couchbase, etc repos for dealing with this.

Reply:
fixed.